content/guides/dolthub-api/hooks.md

```diff
@@ -0,0 +1,119 @@
+| ---------------------| --------| -------------------------------------------------------------------------------|
+| `ref`                | string  | The name of the branch that was pushed.                                        |
+| `head`               | string  | The SHA of the most recent commit on `ref` after the push.                     |
+| `prev`               | string  | The SHA of the most recent commit on `ref` before the push.                    |
+| `repository.name`    | string  | The name of the repository where the push occurred.                            |
+| `repository.owner`   | string  | The username of the owner of the repository where the push occurred.           |
+
+An example of the payload:
+
+```json
+{
+  "ref": "refs/heads/main",
+  "head": "ns725d8noah3m0mjjvrilet1rsmcgna2",
+  "prev": "6higvr7ic9ndahfruh3kufu409im44jd",
+  "repository": {
+    "name": "test",
+    "owner": "liuliu"
+  }
+}
+```
+
+### on_pull_request
+
+The Pull Request event webhook is triggered whenever a pull request is created, merged, or closed, the payload includes these fields:
+
+| Key                  | Type    | Description                                                                   |
+| ---------------------| --------| ------------------------------------------------------------------------------|
+| `repository.name`    | string  | The name of the repository where the push occurred.                           |
+| `repository.owner`   | string  | The username of the owner of the repository where the push occurred.          |
+| `pullID`             | string  | The ID of the pull request that triggered the webhook.                        |
+| `action`             | string  | The action that triggered the webhook (Opened, Closed, etc.).                 |
+
+An example of the payload object:
+
+```json
+{
+  "repository": {
+    "name": "test",
+    "owner": "liuliu"
+  },
+  "pullID": "15",
+  "action": "Opened"
+}
+```
+
+### on_branch
+
+A Branch Event Webhook is triggered when a branch is created or deleted, the payload includes these fields:
+
+| Key                  | Type    | Description                                                                   |
+| ---------------------| --------| ------------------------------------------------------------------------------|
+| `repository.name`    | string  | The name of the repository where the push occurred.                           |
+| `repository.owner`   | string  | The username of the owner of the repository where the push occurred.          |
+| `event_name	`        | string  | The name of the event (branch).                                               |
+| `branch`             | string  | The full name of the branch (e.g., refs/heads/main).                          |
+| `action`             | string  | The action that triggered the webhook (deleted or created the branch).        |
+
+An example of the payload:
+```json
+{
+  "event_name": "branch",
```

Comment:
Why does this have an event name and the others don't?

Reply:
i added this `event_name` to the branch payload in dolthubapi, I didn't realize the others don't have it, and didn't want to change the existing ones to affect customer use..